### ISSUE TITLE
link failure on platforms that require libiconv

### ISSUE THREAD

**Issue by grobian** [rhash/RHash#243]
while running configure:
```
Checking for sources ... RHash 1.4.4
Checking for target OS ... Darwin
Checking for arm64-apple-darwin22-gcc version ... gcc 12.2.0
Checking for gettext ... found
Checking for OpenSSL ... found
./configure: line 879: -lintl: command not found
Writing config.mak
Writing librhash/config.mak
Writing dist/librhash.pc
```
during make:
```
make[1]: Leaving directory '/Users/fabian/Gentoo-13/var/tmp/portage/app-crypt/rhash-1.4.4/work/RHash-1.4.4-.arm64/librhash'
arm64-apple-darwin22-gcc calc_sums.o hash_print.o common_func.o hash_update.o file.o file_mask.o file_set.o find_file.o hash_check.o output.o parse_cmdline.o rhash_main.o win_utils.o librhash/librhash.1.dylib  -Wl,-dead_strip_dylibs -o rhash
Undefined symbols for architecture arm64:
  "_libintl_bindtextdomain", referenced from:
      _main in rhash_main.o
  "_libintl_gettext", referenced from:
      _rename_file_by_embeding_crc32 in calc_sums.o
      _save_torrent_to in calc_sums.o
      _print_sfv_banner in hash_print.o
      _fix_sfv_header in hash_update.o
      _update_ctx_free in hash_update.o
      _process_hash_file in hash_check.o
      _check_embedded_crc32 in hash_check.o
      ...
  "_libintl_setlocale", referenced from:
      _main in rhash_main.o
  "_libintl_textdomain", referenced from:
      _main in rhash_main.o
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status
make: *** [Makefile:153: rhash] Error 1
```

Problem seems to occur because configure does an unquoted assignment:

```
if test "$OPT_STATIC" = "no"; then
  LIBRHASH_TYPE=shared
  LIBRHASH_PATH="\$(LIBRHASH_SHARED)"
  RHASH_TEST_OPTIONS=--shared
  test "$INSTALL_LIB_SHARED" = "auto" && INSTALL_LIB_SHARED=yes
  test "$INSTALL_LIB_STATIC" = "yes"  &&                                        RHASH_BUILD_TARGETS="$RHASH_BUILD_TARGETS \$(LIBRHASH_STATIC)"
  RHASH_LDFLAGS=$RHASH_LDFLAGS $GETTEXT_LDFLAGS  <------- here
else
```
That needs to be quoted
```
RHASH_LDFLAGS="$RHASH_LDFLAGS $GETTEXT_LDFLAGS"
```

**Comment by rhash** [comment]
From what shell do you run the configure script?

**Comment by grobian** [comment]
```
$ foo=bar baz
bash: baz: command not found
$ echo $BASH_VERSION
5.2.15(1)-release
```

**Comment by rhash** [comment]
Thank you for reporting this bug! It is fixed by c804982d1b24c6533118b3068d2cfdca685076f3.
